src/libstore/profiles.hh

```diff
@@ -37,6 +37,8 @@ void deleteGeneration(const Path & profile, unsigned int gen);
 
 void deleteGenerations(const Path & profile, const std::set<unsigned int> & gensToDelete, bool dryRun);
 
+void deleteGenerationsGreaterThan(const Path & profile, const int max, bool dryRun);
```

Comment:
Can `deleteGenerations` be removed? It looks like a special case of `deleteGenerationsGreaterThan` with `max=1`.

Reply:
No, ```deleteGenerations``` can be given a list of specific generations where as mine will only keep x in chronological order, or the date one will only keep so many dates back.

Reply:
@edolstra any chance we can get this merged before 18.03 release?

Reply:
I suppose we could unify the two functions by simply having `deleteGenerationsOlderThan` build the `std::set` and then pass it into `deleteGenerations`. I doubt the deduplication would be that valuable though.

Although I did notice that you don't check against `curGen`, but `deleteGenerations` does. Do you do anything to ensure you don't delete the current generation? If not, perhaps it'd pay to just call the existing function with the set you computed.

Reply:
@copumpkin I could be mistaken but rereading the logic I wrote, it shouldn't be possible to delete the current generation, the list of gens checks to see if it hits the end of generations and bails out at that point.

Reply:
@copumpkin you were correct that it would have been possible to have deleted the currentgeneration if you had done enough rollbacks. I have modified the code to make sure this is not possible